src/simple-fetch.ts

```diff
@@ -0,0 +1,18 @@
+import type {FetchResponse} from '@rocicorp/licensing/src/client';
+
+// This wrapper around fetch adapts it to the narrow fetch interface
+// that the licensing client expects. That type enables the licensing
+// client to work with both the browser and node-fetch implementations.
+export async function browserSimpleFetch(
```

Comment:
Why not just narrow the type instead of having a runtime wrapper??

If you really want to be this precise I recommend defining the following type:

```ts
type SimpleResponse = {
  status: number;
  text: () => Promise<string>;
};

type SimpleFetch = (
  url: string,
  options: {
    method: string;
    body: string | null;
    headers: string[][];
  }
) => Promise<SimpleResponse>;
```

(or all it FetchSubset or something like that)

Then you can directly pass `fetch` and `nodeFetch` without a runtime wrapper

Reply:
My understanding was that there was no useful common subset of fetch between the browser and node ones...

Reply:
The above types cover the usages and there is nothing in the above type that is would expose those underlying differences.